### DISCUSSION TITLE
Use Trunk instead of Leptos CLI as starter?

### DISCUSSION THREAD

**Discussion by floric** [leptos-rs/leptos#190] [General]
Hey :)

at first thanks for this awesome framework. I'm currently exploring it and I'm really happy so far.
As part of my journey I started with the starter project and was directly overwhelmed by its complexity including SSR etca. Then I went back and used the Counter example with CSR and Trunk which is a much easier start and usually enough to get started for simple use cases and tests (and also more comparable to other JS based SPA starters without SSR).

It seems like using the Leptos CLI or Trunk are two complete ways right now, regarding their tooling and complexity. I had hoped to use the Leptos CLI for this simple project like a wrapper of Trunk. Does this make sense? For now I'm fine without the Leptos CLI at all and use only Trunk. But maybe this big difference makes it difficult for other learners to get started as the starter project requires much more understanding of a lot of concepts like SSR, hydration, nightly/stable Rust releases. So just a short notice about my experience so far. :)

**Comment by gbj** [comment]
Hi! Great to hear you've been enjoying it!

At the moment, here's how to think of it:
- Trunk is the best tool if you're just trying to get going with a simple, browser-only ("client-side") app and learn the library.
- `cargo-leptos` is *by far* the easiest way to get started with a batteries-included integrated server-client setup, with parallelized live-reloading builds, SASS support, etc.. We've been in a process of simplifying the project structure and then using it for the other examples in the repo, so the complexity should diminish over time a bit. Even as things are now this is a *crazy* powerful tool and much simpler than trying to do this client-server setup manually; but still more complex than "hey just run some stuff in the browser."

Thanks for your feedback!

**Reply by floric** [reply]
Ok :) Then at least I understood it correctly. Thanks for your response.
